web/frontend/src/components/slam.vue

```diff
@@ -95,6 +105,84 @@ const fetchSLAMPose = (name: string): Promise<commonApi.Pose> => {
   });
 };
 
+const executeMoveOnMap = async () => {
+  moveClick = false;
+
+  /*
+   * set request name
+   * here we set the name of the motion service the user is using
+   */
+  motionServiceReq.setName('builtin');
+
+  const value = await fetchSLAMPose(props.name);
+  // set pose in frame
+  const destination = new commonApi.Pose();
+  destination.setX(Math.abs(destinationMarker.x - value.getX()));
+  destination.setY(Math.abs(destinationMarker.z - value.getY()));
+  destination.setZ(value.getZ());
+  destination.setOX(value.getOX());
+  destination.setOY(value.getOY());
+  destination.setOZ(value.getOZ());
+  destination.setTheta(value.getTheta());
+  motionServiceReq.setDestination(destination);
+
+  // set SLAM resource name
+  const slamResourceName = new commonApi.ResourceName();
+  slamResourceName.setNamespace('rdk');
+  slamResourceName.setType('service');
+  slamResourceName.setSubtype('slam');
+  slamResourceName.setName(props.name);
+  motionServiceReq.setSlamServiceName(slamResourceName);
+
+  // set component name
+  const baseResources = filterResources(props.resources, 'rdk', 'component', 'base');
+  if (baseResources === undefined) {
```

Comment:
validation moved to the top of the function.

`if (baseResources === undefined || baseResources.length > 0)` wouldn't this then always be true?

the question of whether is it possible to have more than one base is interesting.
supposing we define the builtin slam service and another one named "special".
we can then have two rovers, each of which used a different slam. 
In this case, `filterResources(props.resources, 'rdk', 'component', 'base');` would return an array of length two, but we have no way of knowing which base is using which slam.

for this reason, ive listed the assumption that we are only rendering one base.

Reply:
Sorry, the if statement I recommended was wrong. 
I think it is better to error out if there is more than one base with an informative error message rather than having it be an implicit assumption.